android/src/main/java/com/auth0/react/A0Auth0Module.java

```diff
@@ -53,26 +54,23 @@ public String getName() {
     @ReactMethod
     public void showUrl(String url, boolean closeOnLoad, Callback callback) {
         final Activity activity = getCurrentActivity();
+        final Uri parsedUrl = Uri.parse(url);
         this.callback = callback;
 
         try {
-            if (activity != null) {
-                CustomTabsIntent.Builder builder = new CustomTabsIntent.Builder();
-                CustomTabsIntent customTabsIntent = builder.build();
-                customTabsIntent.launchUrl(activity, Uri.parse(url));
-            } else {
-                final Intent intent = new Intent(Intent.ACTION_VIEW);
-                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-                intent.setData(Uri.parse(url));
-                getReactApplicationContext().startActivity(intent);
+            if (activity != null) AuthenticationActivity.authenticateUsingBrowser(activity, parsedUrl);
+            else {
+                final WritableMap error = Arguments.createMap();
+                error.putString("error", "a0.activity_not_available");
+                error.putString("error_description", "Android Activity is null.");
```

Comment:
I think it's a failure scenario in which the dev cannot do anything to recover from it after receiving the error in the callback. Did you run into this issue during your tests?  Would it make sense to just log the error and ignore launching the authentication instead?

The "browser app missing" error is already handled below, that's the important one. Thoughts?

Reply:
This is not an error to recover from, it's an error to run into during development and then take the appropriate measures to prevent it (e.g. by making sure there is an activity). I think failing silently is not a desirable outcome.

Reply:
See https://github.com/react-native-google-signin/google-signin/blob/master/android/src/main/java/co/apptailor/googlesignin/RNGoogleSigninModule.java#L189 for a similar example.

Reply:
By making it an error, the dev can send it to Crashlytics if it ever happens on production for some reason.

Reply:
Yeah, I raised it because we require an "activity context", and other types of contexts such as an "application context" wouldn't work. We need an activity to launch another activity and await a result from it.

Thanks for the link, is interesting to see that other devs also do it. Let's keep it then.